cores/esp8266/Arduino.h

```diff
@@ -127,21 +128,26 @@ void timer0_isr_init(void);
 void timer0_attachInterrupt(timercallback userFunc);
 void timer0_detachInterrupt(void);
 
-// undefine stdlib's abs if encountered
+// undefine stdlib's definitions when encountered, provide abs that supports floating point for C code
+// in case we are using c++, these will either be:
+// - undef'ed by the algorithm include down below, implicitly including cstdlib
+// - undef'ed by the stdlib.h header up above in a more recent versions of gcc
 #ifdef abs
 #undef abs
+#define abs(x) ((x)>0?(x):-(x))
 #endif
 
-#define abs(x) ((x)>0?(x):-(x))
+#ifdef round
+#undef round
+#define round(x) ((x)>=0?(long)((x)+0.5):(long)((x)-0.5))
+#endif
```

Comment:
:oops: this is actually wrong as there is no `define round`, missed when editing.
meaning, right now this pr only uses std::round and master uses macro